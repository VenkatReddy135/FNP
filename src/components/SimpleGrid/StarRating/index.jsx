import React from "react";
import Rating from "@material-ui/lab/Rating";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles(() => ({
  ratingContainer: {
    marginTop: "20px",
  },
}));

/**
 * Function to convert rating values into Star rating to display in Grid
 *
 * @name StarRating
 * @param {object} props - Passing record, source
 * @returns {React.ReactElement} returns star rating component
 */
const StarRating = (props) => {
  const { record, source } = props;
  const classes = useStyles();

  return (
    <Rating className={classes.ratingContainer} data-testid="star-rating" defaultValue={record[source]} readOnly />
  );
};

StarRating.propTypes = {
  record: PropTypes.objectOf(PropTypes.any),
  source: PropTypes.string.isRequired,
};
StarRating.defaultProps = {
  record: {},
};

export default StarRating;
