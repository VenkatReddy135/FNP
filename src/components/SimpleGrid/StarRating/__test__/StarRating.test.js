/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import StarRating from "..";

describe(`Star Rating component test cases`, () => {
  const props = {
    record: {
      overallRating: 5,
    },
    source: "overallRating",
  };

  beforeEach(() => {
    render(<StarRating {...props} />);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render Star Rating component", () => {
    expect(screen.getByTestId("star-rating")).toBeTruthy();
  });

  it("should render exact rating", () => {
    const ratingElement = screen.getByTestId("star-rating");
    const labeAttribute = ratingElement.attributes.getNamedItem("aria-label");
    expect(labeAttribute.value).toBe("5 Stars");
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<StarRating {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
