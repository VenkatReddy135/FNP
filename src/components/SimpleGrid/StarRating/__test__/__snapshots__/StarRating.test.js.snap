// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star Rating component test cases should match snapshot 1`] = `
<span
  aria-label="5 Stars"
  className="MuiRating-root MuiRating-readOnly"
  data-testid="star-rating"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconFilled"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;
